web/src/app/chat/components/input/ActionManagement.tsx

```diff
@@ -439,79 +481,255 @@ function MCPToolsList({
+              }
+              readOnly
+              onClick={
+                allToolsDisabled
+                  ? enableAllToolsForServer
+                  : disableAllToolsForServer
+              }
+              className="
+                w-full
+                pl-7
+                pr-9
+                py-2
+                bg-transparent
+                rounded-lg
+                text-sm
+                outline-none
+                text-neutral-700 dark:text-neutral-300
+                placeholder:text-neutral-400 dark:placeholder:text-neutral-500
+                cursor-pointer
+              "
+            />
+            <img
+              src={allToolsDisabled ? "/plug.svg" : "/unplug.svg"}
+              alt={allToolsDisabled ? "Enable All Tools" : "Disable All Tools"}
+              className="absolute right-3 top-1/2 transform -translate-y-1/2 w-4 h-4 pointer-events-none"
+              style={{
+                filter:
+                  theme === "dark"
+                    ? "invert(0.6) sepia(1) saturate(0) hue-rotate(0deg) brightness(1.2)"
+                    : "invert(0.6) sepia(1) saturate(0) hue-rotate(0deg) brightness(0.6)",
+                opacity: theme === "dark" ? 0.9 : 0.8,
+              }}
+            />
+          </div>
+        </div>
+        <div className="border-b border-border mx-1 mt-1" />
+        <div
+          className="mx-1 h-2 -mb-2 transition-opacity ease-out"
+          style={{
+            background:
+              "linear-gradient(to bottom, rgba(0, 0, 0, 0.06), transparent)",
+            opacity: showTopShadow ? 1 : 0,
+          }}
+        />
       </div>
 
-      {/* Tools List */}
-      <div className="pt-2 flex-1 overflow-y-auto mx-1 pb-2">
-        {filteredTools.length === 0 ? (
-          <div className="text-center py-1 text-text-400">
-            No matching tools found
+      <div
+        id="action-submenu-scroll"
+        className="flex-1 overflow-y-auto min-h-0 relative"
+        onScroll={(e) => onScrollStateChange(e.currentTarget)}
+      >
+        <div className="space-y-1.5 pb-2 pt-2">
+          {filteredTools.length === 0 ? (
+            <div className="text-center py-4 text-neutral-400 dark:text-neutral-500">
+              No matching tools found
+            </div>
+          ) : (
+            filteredTools.map((tool) => {
+              const isDisabled = disabledToolIds.includes(tool.id);
+              const isEnabled = !isDisabled;
+
+              const label = (
+                <div className="flex flex-col cursor-default">
+                  <span
+                    className={`text-sm font-medium ${
+                      isEnabled
+                        ? "text-neutral-700 dark:text-neutral-200"
+                        : "text-neutral-400 dark:text-neutral-500"
+                    }`}
+                  >
+                    {tool.display_name || tool.name}
+                  </span>
+                </div>
```

Comment:
Is this a "tag"? Or just a simple text component? If the former is true, please try using the `Tag` component. If the latter is true, please render using the refreshed `Text` component, please!